util/genrest/protomodelcreator.go

```diff
@@ -141,3 +167,108 @@ func NewRESTRequestPattern(rule *annotations.HttpRule) (*protomodel.RESTRequestP
+	_ = plugin
+
+	y, err := ioutil.ReadFile(serviceConfigPath)
+	if err != nil {
+		cwd, _ := os.Getwd()
+		return nil, fmt.Errorf("error reading service config %q (cwd==%q): %v", serviceConfigPath, cwd, err)
+	}
+
+	j, err := yaml.YAMLToJSON(y)
+	if err != nil {
+		return nil, fmt.Errorf("error converting YAML to JSON: %v", err)
+	}
+
+	serviceConfig := &serviceconfig.Service{}
+	if err := (protojson.UnmarshalOptions{DiscardUnknown: true}).Unmarshal(j, serviceConfig); err != nil {
+		return nil, fmt.Errorf("error unmarshaling service config: %v", err)
+	}
+
+	// An API Service Config will always have a `name` so if it is not populated,
+	// it's an invalid config.
+	if serviceConfig.GetName() == "" {
+		return nil, fmt.Errorf("invalid API service config file %q", serviceConfigPath)
+	}
+	return serviceConfig, nil
+}
+
+// Mixins is the collection of files containing methods to be mixed in.
+type Mixins []*MixinFile
+
+// MixinFile describes a single file containins methods to be mixed in.
+type MixinFile struct {
+	file     *descriptor.FileDescriptorProto
+	services []*MixinService
+}
+
+// MixinService describes a single service containing methods to be filled in
+type MixinService struct {
+	service *descriptor.ServiceDescriptorProto
+	methods []*descriptor.MethodDescriptorProto
+}
+
+// indexedRules keys HTTP rules by their selectors
+type indexedRules map[string]*annotations.HttpRule
+
+// collectMixins collects the configured mixin APIs from the Service config and
+// gathers the appropriately configured mixin methods to generate for each.
+func collectMixins(serviceConfig *serviceconfig.Service) Mixins {
+	mixinRules := indexedRules{}
+	for _, rule := range serviceConfig.GetHttp().GetRules() {
+		mixinRules[rule.GetSelector()] = rule
+	}
+	mixins := Mixins{}
+	for _, api := range serviceConfig.GetApis() {
+		if _, ok := mixinDescriptors[api.GetName()]; ok {
+			mixins = append(mixins, collectMixinMethods(mixinRules, api.GetName())...)
+		}
+	}
+	return mixins
+}
+
+func collectMixinMethods(mixinRules indexedRules, api string) Mixins {
+	files := Mixins{}
+	for _, file := range mixinDescriptors[api] {
+		fileToAdd := &MixinFile{
+			file: file,
+		}
+		files = append(files, fileToAdd)
+		for _, service := range file.GetService() {
+			serviceToAdd := &MixinService{
+				service: service,
+			}
+			fileToAdd.services = append(fileToAdd.services, serviceToAdd)
+			for _, method := range service.GetMethod() {
+				fqn := fmt.Sprintf("%s.%s.%s", file.GetPackage(), service.GetName(), method.GetName())
+
+				if rule := mixinRules[fqn]; rule != nil {
+					proto.SetExtension(method.Options, annotations.E_Http, rule)
+					serviceToAdd.methods = append(serviceToAdd.methods, method)
+
+				}
+			}
+		}
+	}
+	return files
+}
+
+var mixinDescriptors map[string][]*descriptor.FileDescriptorProto
+
+func init() {
+	mixinDescriptors = map[string][]*descriptor.FileDescriptorProto{
+		"google.longrunning.Operations": {
```

Comment:
Note to self: add the other mixin services too (Locations, IAMPolicy)

Reply:
I was getting some (probably trivial) errors with Locations, so I just excluded the others for now so we could get LRO working. But as you see, I copied your structure from gapic-generator-go so the others would be easy to add back in.